src/types/profile.js

```diff
@@ -211,6 +213,20 @@ export type PausedRange = {
   reason: 'profiler-paused' | 'collecting',
 };
 
+export type JsTracerTable = {
+  events: JsTracerEvents,
+  stringTable: UniqueStringArray,
+};
+
+export type JsTracerEvents = {
+  events: Array<IndexIntoJsTracerStringTable>,
+  timestamps: Array<Microseconds>,
+  durations: Array<Microseconds | -1>,
+  lines: Array<number | -1>, // Line number.
+  columns: Array<number | -1>, // Column number.
+  length: number,
+};
```

Comment:
Ah good point on the exact object.

I'm a little hesitant to add "read only" to these, mainly because none of the other profile data structures enforce this, and we occasionally mutate values here and there when processing and writing test data structures. I would prefer to follow-up with a discussion outside of this PR if we want to pursue this, as it's something we are not currently doing.